File: Demizon.Dal/Entities/VideoLink.cs
namespace Demizon.Dal.Entities;

public class VideoLink
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;

    public bool IsVisible { get; set; } = false;

    public string Url { get; set; } = null!;

    public int Year { get; set; }

    public bool IsInternal { get; set; } = false;

    public int? DanceId { get; set; }

    public virtual Dance? Dance { get; set; }
}

File: demizon_flutter/lib/models/video_link.dart
import 'package:json_annotation/json_annotation.dart';

part 'video_link.g.dart';

/// Protějšek Demizon.Contracts/Dances/VideoLinkDto.cs
///
/// Odkaz na video u tance. `isInternal` znamená, že video je jen pro členy.
@JsonSerializable()
class VideoLink {
  const VideoLink({
    required this.id,
    required this.name,
    required this.url,
    required this.year,
    this.isVisible = true,
    this.isInternal = false,
  });

  final int id;
  final String name;
  final String url;
  final int year;
  final bool isVisible;
  final bool isInternal;

  factory VideoLink.fromJson(Map<String, dynamic> json) =>
      _$VideoLinkFromJson(json);

  Map<String, dynamic> toJson() => _$VideoLinkToJson(this);
}
